qiskit/quantum_info/states/stabilizerstate.py

```diff
@@ -259,7 +259,34 @@ def evolve(
         ret._data = self.clifford.compose(other.clifford, qargs=qargs)
         return ret
 
-    def expectation_value(self, oper: Pauli, qargs: None | list = None) -> complex:
+    def expectation_value(self, oper: Pauli | SparsePauliOp, qargs: None | list = None) -> complex:
+        """Compute the expectation value of a Pauli or SparsePauliOp operator.
+
+        Args:
+            oper (Pauli or SparsePauliOp): a Pauli or SparsePauliOp operator to evaluate expval.
+            qargs (None or list): subsystems to apply the operator on.
+
+        Returns:
+            complex: the expectation value.
+
+        Raises:
+            QiskitError: if oper is not a Pauli or SparsePauliOp operator.
+        """
+        if isinstance(oper, Pauli):
+            return self._expectation_value_pauli(oper, qargs)
+
+        if isinstance(oper, SparsePauliOp):
+            return sum(
+                coeff * self._expectation_value_pauli(Pauli((z, x)), qargs)
+                for z, x, coeff in zip(oper.paulis.z, oper.paulis.x, oper.coeffs)
+            )
+
+        else:
+            raise QiskitError(
```

Comment:
`TypeError` might be more appropriate (but if this is Qiskit's convention then it's fine)

Reply:
I see that also previously it was a `QiskitError`:

https://github.com/Qiskit/qiskit/blob/1263d82877ba5183232db9bac5bc2c61a7da7a26/qiskit/quantum_info/states/stabilizerstate.py#L272